newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -10,13 +10,59 @@ import {
   getNewdle,
   getNumberOfAvailableAnswers,
   getNumberOfTimeslots,
+  getParticipant,
   isAllAvailableSelected,
   isAllAvailableSelectedImplicitly,
 } from '../../answerSelectors';
-import {chooseAllAvailable} from '../../actions';
+import {abortAnswering, chooseAllAvailable, fetchParticipant} from '../../actions';
 import styles from './answer.module.scss';
 import client from '../../client';
 
+function ParticipantName({anonymous, setName, onSubmit, disabled}) {
+  const dispatch = useDispatch();
+  const {partcode: participantCode, code: newdleCode} = useParams();
+  const participant = useSelector(getParticipant);
+
+  useEffect(() => {
+    if (!anonymous) {
+      dispatch(fetchParticipant(newdleCode, participantCode));
+      return () => {
+        dispatch(abortAnswering());
+      };
+    }
+  }, [dispatch, newdleCode, participantCode, anonymous]);
+
+  if (anonymous) {
+    return (
+      <div className={styles['participant-name-box']}>
+        <h3>Who are you?</h3>
+        <Input
+          autoFocus
+          transparent
+          className={styles['participant-name-input']}
+          placeholder="Please enter your name..."
+          disabled={disabled}
+          onChange={(_, data) => setName(data.value)}
+          onKeyDown={e => {
+            if (e.key === 'Enter') {
+              onSubmit();
+            }
+          }}
+        />
+      </div>
+    );
+  } else {
+    return (
+      participant && (
```

Comment:
Can it be `null` or `undefined` if it is for sure not `anonymous`?

Reply:
Yes, while it's being loaded.

Reply:
Ah, right.